src/node/state/memory.rs

```diff
@@ -101,4 +103,15 @@ impl State for MemoryState {
     ) -> Result<Option<Cookie>, anyhow::Error> {
         Ok(self.cookies.get(&socket_addr).map(|c| c.to_owned()))
     }
+
+    async fn add_peers(&mut self, addresses: Vec<SocketAddr>) -> Result<(), anyhow::Error> {
```

Comment:
Reminder: this needs tests to ensure if a peer appears twice it will be only added once. This is enforced in the memory state by a `HashSet` but might need a deduper in the real thing.

Reply:
Sled is a key value store so should be able to deduplicate. I haven't used it enough to know what happens if something is inserted twice with the same key. 🤷